src/WebUI/Workloads/WorkloadsPivot.tsx

```diff
@@ -107,15 +104,18 @@ export class WorkloadsPivot extends React.Component<IWorkloadsPivotProps, IWorkl
     private _onDataFound = (): void => {
         const workloadSize = this._workloadsStore.getWorkloadSize();
         if (this.state.workloadResourceSize <= 0 && workloadSize > 0) {
-            this.setState({ workloadResourceSize: workloadSize });
+            this.setState({ workloadResourceSize: workloadSize }, () => {
+                // resettting counts upon workloads details load
+                this._childNodes = 0;
```

Comment:
here also we can follow dictionary mode, as callers are calling only once. so we can have deployment and other kind of key and make it true, based on the key that we receive in the notifyrender. then check for all keys, if dict[key] is true then we are good.

Reply:
Done. Used a dictionary to track the metadata loading in `OtherWorkloads` page and another dictonary to track child components to be in ready state in `WorkloadsPivot`